### ISSUE TITLE
xmpp broken?

### ISSUE THREAD

**Issue by andig** [containrrr/shoutrrr#57]
Running `go get -u` I'm getting:

```
# github.com/containrrr/shoutrrr/pkg/xmpp
../../go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200810085354-a9206f031449/pkg/xmpp/xmpp.go:36:31: not enough arguments in call to xmpp.NewClient
        have (xmpp.Config, *xmpp.Router)
        want (*xmpp.Config, *xmpp.Router, func(error))
../../go/pkg/mod/github.com/containrrr/shoutrrr@v0.0.0-20200810085354-a9206f031449/pkg/xmpp/xmpp_config.go:112:3: unknown field 'Password' in struct literal of type xmpp.Config
```

**Comment by simskij** [comment]
@piksel maybe?

**Comment by piksel** [comment]
Yeah, XMPP is not completed, sorry.

**Comment by andig** [comment]
> Yeah, XMPP is not completed, sorry. See #6

I understand, but it would still be nice if the package was updatable and compiled? Keep work in progress inside a PR branch instead of mainline until at least compiling would help.

**Comment by piksel** [comment]
@andig it compiles if you don't change the dependencies. `go get -u` upgrades `gosrc.io/xmpp` from `v0.1` to `v0.5`, which is why you are getting the error.

```
nils @ Ketsueki ~/dev/shoutrrr [master ?:1 ✗]
✔ ➡ go build

nils @ Ketsueki ~/dev/shoutrrr [master ?:1 ✗]
✔ ➡ go get -u
go: github.com/mattn/go-isatty upgrade => v0.0.12
go: github.com/onsi/ginkgo upgrade => v1.14.0
go: gosrc.io/xmpp upgrade => v0.5.1
go: golang.org/x/sys upgrade => v0.0.0-20200812155832-6a926be9bd1d
go: github.com/mattn/go-colorable upgrade => v0.1.7
go: github.com/fatih/color upgrade => v1.9.0
go: golang.org/x/xerrors upgrade => v0.0.0-20200804184101-5ec99f83aff1
go: github.com/fsnotify/fsnotify upgrade => v1.4.9
go: nhooyr.io/websocket upgrade => v1.8.6
go: github.com/klauspost/compress upgrade => v1.10.11
# github.com/containrrr/shoutrrr/pkg/xmpp
pkg/xmpp/xmpp.go:36:31: not enough arguments in call to xmpp.NewClient
        have (xmpp.Config, *xmpp.Router)
        want (*xmpp.Config, *xmpp.Router, func(error))
pkg/xmpp/xmpp_config.go:112:3: unknown field 'Password' in struct literal of type xmpp.Config

nils @ Ketsueki ~/dev/shoutrrr [master U:2 ?:1 ✗]
╳ ➡ go build
# github.com/containrrr/shoutrrr/pkg/xmpp
pkg/xmpp/xmpp.go:36:31: not enough arguments in call to xmpp.NewClient
        have (xmpp.Config, *xmpp.Router)
        want (*xmpp.Config, *xmpp.Router, func(error))
pkg/xmpp/xmpp_config.go:112:3: unknown field 'Password' in struct literal of type xmpp.Config
```

**Comment by simskij** [comment]
> > Yeah, XMPP is not completed, sorry. See #6
> 
> I understand, but it would still be nice if the package was updatable and compiled? Keep work in progress inside a PR branch instead of mainline until at least compiling would help.

As @piksel pointed out, it works if you keep to the checked-in package versions.

I'm not looking to be an ass here, but this is an open-source project, without any substantial financial backing. As such, it would be greatly appreciated if you didn't dive in here and started making demands on how to set up our development pipeline. Even if you would have been right in the package not being compilable.

You're more than welcome to contribute with a PR if you find something that you'd like to change or improve. 👍 

Thanks.

**Comment by andig** [comment]
No harshness intended and I would appreciate the same ;) I didn't mean to make demands, instead I was under the impression that HEAD is broken. I've realized my mistake...

**Comment by piksel** [comment]
Still, due to how `go get` works this got really tedious, so I fixed it in #122.
